methods.rb

```diff
@@ -1 +1,23 @@
-# Your code here!
+def greet_programmer
+  puts "Hello, programmer!"
+end
+
+def greet(name)
+  puts "Hello, #{name}!"
+end
+
+def greet_with_default(name = "programmer")
+  puts "Hello, #{name}!"
+end
+
+def add (num1, num2)
+  return num1 + num2
+end
+
+def halve(num)
+  if num.class != Integer
+    return nil
+  end
+
+  return num / 2
```

Comment:
The final line of a ruby method is always returned. No need to explicitly return here. 
```suggestion
  num / 2
```